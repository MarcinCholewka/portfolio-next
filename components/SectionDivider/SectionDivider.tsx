'use client';

import React from 'react';
import { motion } from 'framer-motion';

export const SectionDivider = () => {
  return (
    <motion.div
      className='bg-violet-300 my-16 h-16 w-1 rounded-full hidden sm:block'
      data-testid='section-divider'
      initial={{ opacity: 0, y: 100 }}
      animate={{ opacity: 1, y: 0 }}
      transition={{ delay: 0.125 }}></motion.div>
  );
};
