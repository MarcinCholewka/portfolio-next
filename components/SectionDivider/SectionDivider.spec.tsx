import React from 'react';
import { render } from '@testing-library/react';

import { SectionDivider } from '@/components/SectionDivider';

import '@testing-library/jest-dom';

describe('SectionDivider', () => {
  it('renders without crashing', () => {
    const { container, getByTestId } = render(<SectionDivider />);
    const divider = getByTestId('section-divider');

    expect(container).toMatchSnapshot();
    expect(divider).toBeInTheDocument();
  });
});
