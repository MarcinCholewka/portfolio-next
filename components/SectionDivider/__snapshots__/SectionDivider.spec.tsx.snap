// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionDivider renders without crashing 1`] = `
<div>
  <div
    class="bg-violet-300 my-16 h-16 w-1 rounded-full hidden sm:block"
    data-testid="section-divider"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  />
</div>
`;
